File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/AddOpeningHoursTest.java
package de.westnordost.streetcomplete.quests;

import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryAdd;
import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
import de.westnordost.streetcomplete.quests.AOsmElementQuestTypeTest;
import de.westnordost.streetcomplete.quests.opening_hours.AddOpeningHours;
import de.westnordost.streetcomplete.quests.opening_hours.AddOpeningHoursForm;

import static org.mockito.Mockito.mock;

public class AddOpeningHoursTest extends AOsmElementQuestTypeTest
{
	@Override protected OsmElementQuestType createQuestType()
	{
		return new AddOpeningHours(mock(OverpassMapDataDao.class));
	}

	public void testOpeningHours()
	{
		bundle.putString(AddOpeningHoursForm.OPENING_HOURS, "my cool opening hours");
		verify(new StringMapEntryAdd("opening_hours", "my cool opening hours"));
	}
}

File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/AddOpeningHoursTest.kt
package de.westnordost.streetcomplete.quests

import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryAdd
import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
import de.westnordost.streetcomplete.quests.opening_hours.AddOpeningHours
import de.westnordost.streetcomplete.quests.opening_hours.AddOpeningHoursForm
import org.junit.Test

import org.mockito.Mockito.mock

class AddOpeningHoursTest : AOsmElementQuestTypeTest() {

    override val questType = AddOpeningHours(mock(OverpassMapDataDao::class.java))

    @Test fun openingHours() {
        bundle.putString(AddOpeningHoursForm.OPENING_HOURS, "my cool opening hours")
        verify(StringMapEntryAdd("opening_hours", "my cool opening hours"))
    }

    @Test fun noOpeningHoursSign() {
        bundle.putBoolean(AddOpeningHoursForm.NO_SIGN, true)
        verify(StringMapEntryAdd("opening_hours:signed", "no"))
    }
}
